modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1418,6 +1428,69 @@ For verifying that an expression is not of a certain type, the following alterna
 <expr> IS NOT TYPED <TYPE>
 ----
 
+[[type-predicate-expressions-any-and-nothing]]
+=== Use of Any and Nothing types
+
+The `ANY` type is a supertype of all types and will therefore match on everything.
```

Comment:
```suggestion
The `ANY` type is a supertype which matches any type.
```

Reply:
Saying it this way is not very helpful though? I would prefer not to use the name of the time to describe what it matches on, also just saying supertype doesn't imply it is of all types, so that information is also useful to keep :)

Reply:
> Saying it this way is not very helpful though?

Why not? To a reader, the important bit of information is that _use `ANY` -> match any type_. Then, the fact that the supertype is of all types (and honestly even that it _is_ a supertype at all) is an implementation detail that doesn't further help users. To my eyes, what they care about is that

1. `ANY` is a _type_, so I can use it in any type expression
2. `ANY` matches all types, and acts a bit like a type-wildcard.

If you don't agree, what extra valuable information do users get from knowing that it's a _supertype of all types_?

Reply:
One thing about the sentence 
```
The `ANY` type is a supertype which matches any type.
```
that I don't particularly like is that it feels like it says nothing by repeating the word `ANY`. 

Maybe something like
```suggestion
The `ANY` type is a supertype which matches values of all types.
```
instead?

Reply:
Yes, because supertype does not imply all types are below it, it implies at least 1 type is. Number is a supertype of Int and float for example, but it is not a supertype of all types.

It is not generally useful to read a description that uses the same word to describe the word, for example if we used this description for a different type:

The `INTEGER` type is a  supertype which matches integer types.

I have not gained any extra information here. I do not know that Integer matches INT32 and INT64 for example. Describing what a type word means using different words is more useful.

Also technically it doesn't match on a type, it is matching on a value from an expression. New suggestion:

The `ANY` type is a supertype of all Cypher types which evaluates to true for all given expressions.

Reply:
Yes, I like this!

Reply:
> The INTEGER type is a supertype which matches integer types.
I have not gained any extra information here. I do not know that Integer matches INT32 and INT64 for example. 

Definitely, but if the supertype is of _all_ types, there is no ambiguity on what types it includes. I do agree the repeated usage of `any` was wrong, but There's suggestion cleared it :)